### DISCUSSION TITLE
Network Wide TamperMonkey

### DISCUSSION THREAD

**Discussion by gagestinson** [Tampermonkey/tampermonkey#1881] [Ideas]
I am studying Electrical Engineering and am a novice to Computer Networks and stuff like that. I have recently found TamperMonkey and thought it might be useful if it was able to be implemented network wide so it does not need to be installed on each  individual device, and it could (potentially) work on devices that are not able to install TamperMonkey directly.

Is this something that could be done? I am thinking an implementation like Wazuh or PiHole where it could be installed on a different device and the TamperMonkey scripts would run on there before being sent to the device that actually requested the webpage.

Please correct me if this is not possible in this implementation, or if it is not possible at all. I think it could be very useful.

**Comment by 7nik** [comment]
1. AFAIK TM can be installed on all commonly used desktop browsers and some mobile browsers.
2. Likely you can inject scripts with a MITM-like proxy, but some GM API requires extension's power:
- GM.XHR (passing CORS);
- GM.setValue/getValue (especially when the script runs across multiple sites);
- GM.cookie (httpOnly cookie);
- GM.getTab/saveTab.
